misc-tools/import-contest

```diff
@@ -0,0 +1,214 @@
+    for file in files:
+        if os.path.exists(file):
+            if confirm(f"Import {basename}s from {file}?", False):
+                print(f"Importing {basename}s.")
+                type = file.split(".")[1]
+                response = upload_file(f"/users/{basename}s", type, file)
+                print(json.dumps(response, indent=4))
+            else:
+                print(f"Skipping {basename}s import.")
+            any_matched = True
+    if not any_matched:
+        if len(files) == 1:
+            print(f"'{files[0]}' not found, skipping {basename}s import.")
+        else:
+            prefix = [f"'{item}'" for item in files[:-1]]
+            prefix_joined = ", ".join(prefix)
+            last = f"'{files[-1]}'"
+            print(
+                f"Neither {prefix_joined} nor {last} found, skipping {basename}s import.")
+
+
+if len(sys.argv) < 2:
+    usage()
+
+api_url = sys.argv[1]
+
+import_file("group", ["groups.json", "groups.tsv"])
+import_file("organization", ["organizations.json"])
+import_file("team", ["teams.json", "teams2.tsv"])
+import_file("account", ["accounts.json", "accounts.yaml", "accounts.tsv"])
+
+# Contest import is a special case: we can have a contest.yaml and problemset.yaml and then we need
+# to import the combined file. We can also gather the contest ID.
+if os.path.exists("contest.yaml"):
+    combine = False
+    if os.path.exists("problemset.yaml"):
+        combine = True
+        question = "Import contest metadata (from contest.yaml and problemset.yaml)?"
+    else:
+        question = "Import contest metadata (from contest.yaml)?"
+    if confirm(question, True):
+        if combine:
+            with open("combined.yaml", "w") as combined, fileinput.input(["contest.yaml", "problemset.yaml"]) as files:
+                for line in files:
+                    combined.write(line)
+            file = "combined.yaml"
+        else:
+            file = "contest.yaml"
+        print(f"Importing contest.")
+        cid = upload_file("contests", "yaml", file)
+elif os.path.exists("contest.json"):
+    if confirm("Import contest metadata (from contest.json)?", True):
+        print("Importing contest.")
+        cid = upload_file("contests", "json", "contest.json")
+else:
+    print("Neither 'contest.yaml' nor 'contest.json' found, skipping contest metadata import.")
+
+if cid is not None:
+    print(f" -> cid={cid}")
+
+# Problem import is also special: we need to upload each individual problem and detect what they are
+if os.path.exists("problems.yaml") or os.path.exists("problems.json") or os.path.exists("problemset.yaml"):
+    if confirm("Import problems?", True):
+        # Check if our user is linked to a team
+        user_data = do_api_request("user")
+        if not "team" in user_data and not confirm("No team associated with your account. Jury submissions won't be imported. Really continue?", False):
+            exit(2)
+
+        print("Importing problems.")
+
+        if cid is None:
```

Comment:
I would move this up to line 162

Reply:
Bellow 162 I assume? And isn't it strange to ask for a contest ID if we DON'T have problems(et).{json,yaml}?